services/blockstorage/adapter/filesystem/file_system_persistence.go

```diff
@@ -268,6 +274,7 @@ func (f *BlockPersistence) WriteNextBlock(blockPair *protocol.BlockPairContainer
 		return false, f.bhIndex.getLastBlockHeight(), errors.Wrap(err, "failed to update index after writing block")
 	}
 
+	f.metrics.indexLastUpdateTime.Update(time.Now().Unix())
```

Comment:
this may fail if metrics is nil

Reply:
this is the case with all ONG metrics updates ..